src/cmd/support_archive/logs.go

```diff
@@ -0,0 +1,123 @@
+package support_archive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const logCollectorName = "logCollector"
+
+type logCollector struct {
+	collectorCommon
+
+	context context.Context
+	pods    v1.PodInterface
+}
+
+func newLogCollector(context context.Context, log logr.Logger, supportArchive tarball, pods v1.PodInterface) collector { //nolint:revive // argument-limit doesn't apply to constructors
+	return logCollector{
+		collectorCommon: collectorCommon{
+			log:            log,
+			supportArchive: supportArchive,
+		},
+		context: context,
+		pods:    pods,
+	}
+}
+
+func (collector logCollector) Do() error {
+	logInfof(collector.log, "Starting log collection")
+
+	podList, err := collector.getPodList()
+	if err != nil {
+		return err
+	}
+
+	podGetOptions := metav1.GetOptions{}
+
+	for _, podItem := range podList.Items {
+		pod, err := collector.pods.Get(collector.context, podItem.Name, podGetOptions)
+		if err != nil {
+			logErrorf(collector.log, err, "Unable to get pod info for %s", podItem.Name)
+		} else {
+			collector.collectPodLogs(pod)
+		}
+	}
+	return nil
+}
+
+func (collector logCollector) Name() string {
+	return logCollectorName
+}
+
+func (collector logCollector) getPodList() (*corev1.PodList, error) {
+	listOptions := metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "pod",
+		},
+		LabelSelector: "app.kubernetes.io/name=dynatrace-operator",
+	}
+	podList, err := collector.pods.List(collector.context, listOptions)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return podList, nil
+}
+
+func (collector logCollector) collectPodLogs(pod *corev1.Pod) {
+	for _, container := range pod.Spec.Containers {
+		podLogOpts := corev1.PodLogOptions{
+			Container: container.Name,
+			Follow:    false,
+		}
+		collector.collectContainerLogs(pod, container, podLogOpts)
+
+		podLogOpts.Previous = true
+		collector.collectContainerLogs(pod, container, podLogOpts)
+	}
+}
+
+func (collector logCollector) collectContainerLogs(pod *corev1.Pod, container corev1.Container, logOptions corev1.PodLogOptions) {
+	req := collector.pods.GetLogs(pod.Name, &logOptions)
+	if req == nil {
+		logErrorf(collector.log, errors.Errorf("Unable to retrieve log stream for pod %s, container %s", pod.Name, container.Name), "")
+		return
+	}
+
+	podLogs, err := req.Stream(collector.context)
+
+	if logOptions.Previous && err != nil {
+		// Soften error message for previous pods a bit, so users don't get nervous. Previous pods often just didn't exist.
+		logInfof(collector.log, "%v", err)
+		return
+	} else if err != nil {
+		logErrorf(collector.log, err, "error in opening stream")
+		return
+	}
+
+	defer podLogs.Close()
+
+	fileName := buildLogFileName(pod, container, logOptions)
```

Comment:
how about a folder per pod with all the container logs (and previous) in it?
Not sure how many files we'll end up with, but seems like having them all in the top level may get confusing

Reply:
GOOD IDEA

I had 3 nodes that makes the following amount of files  => 3*4 (csi) + 2*1 (webhook) + 1 (operator) = 15
this is without any agents or activegate, and if I had 10 nodes this number would go up to 43 which is a bit much

Reply:
I'll have to check how easy it is to create folders in a tarball in memory on-the-fly.

Reply:
piece of cake, done!